### ISSUE TITLE
Why is resolve-uri forbidden from resolving against a URI that contains a fragment identifier?

### ISSUE THREAD

**Issue by ndw** [qt4cg/qtspecs#280]
The 3.1 F&O spec says, of `fn:resolve-uri()`:

> A dynamic error is raised [[err:FORG0002](https://www.w3.org/XML/Group/qtspecs/specifications/xpath-functions-31/html/Overview.html#ERRFORG0002)] if $base is not a valid IRI according to the rules of RFC3987, extended with an implementation-defined subset of the extensions permitted in LEIRI, or if it is not a suitable IRI to use as input to the chosen resolution algorithm (for example, if it is a relative IRI reference, if it is a non-hierarchic URI, or **if it contains a fragment identifier**).

(emphasis added by me)

What in the name of all things is that about? I've never noticed that before, and I haven't seen any other API (the URL API in Node and the browser for example) that cares.

`resolve-uri('test.xml', 'http://example.com/path/file.xml#foo')` === `http://example.com/path/test.xml`

I don't see why the presence of a fragment identifier should matter in the least.

e.g.

```
>> let url = new URL("http://example.com/path/file.xml#foo")
>> console.log(url.href)
http://example.com/path/file.xml#foo
>> let resolved = new URL("test.xml", url);
>> console.log(resolved.href)
http://example.com/path/test.xml
```

**Comment by ChristianGruen** [comment]
Good question; I didn’t dare to ask ;·)

**Comment by michaelhkay** [comment]
I confess the words look suspiciously as if I wrote them -- though I'm sure I only did so after a great deal of detailed WG discussion.

OK: here's the explanation. RFC 3986 §5.1 says

`A base URI must conform to the <absolute-URI> syntax rule ([Section 4.3](https://www.rfc-editor.org/rfc/rfc3986#section-4.3)). 
`
and that syntax does not allow a fragment identifier.

It goes on to say: 

> If the  base URI is obtained from a URI reference, then that reference must be converted to absolute form and stripped of any fragment component prior to its use as a base URI.

So there is justification for the rule, but it does seem very harsh. Stripping the fragment identifier from the base URI seems more friendly, and if you hold your head at the right angle when reading it, the spec does seem to allow it.
